batchflow/models/torch/base_mixins.py

```diff
@@ -221,6 +222,197 @@ class OptimalBatchSizeMixin:
 
     For stable measurements, we make `n` iterations of `train`/`predict`, until the memory consumption stabilizes.
     """
+
+    def is_cuda_out_of_memory(self, exception):
+        """ Check if exception is CUDA OOM """
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "CUDA" in exception.args[0]
+            and "out of memory" in exception.args[0]
+        )
+
+    # not sure if it's necessary
+    def is_cudnn_snafu(self, exception):
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED." in exception.args[0]
+        )
+
+    def is_out_of_cpu_memory(self, exception):
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "DefaultCPUAllocator: can't allocate memory" in exception.args[0]
+        )
+
+    def is_oom_error(self, exception):
+        return (self.is_cuda_out_of_memory(exception) or
+                self.is_cudnn_snafu(exception) or
+                self.is_out_of_cpu_memory(exception))
+
+    def garbage_collection_cuda(self):
+        """ Garbage collection Torch (CUDA) memory. """
+        gc.collect()
+        try:
+            # This is the last thing that should cause an OOM error, but seemingly it can.
+            torch.cuda.empty_cache()
+        except RuntimeError as exception:
+            if not self.is_oom_error(exception):
+                # Only handle OOM errors
+                raise
+
+
+    def _compute_optimal_batch_size(self, method='train', inputs=None, targets=None,
```

Comment:
It is quite strange that this method is "private", while it uses "public" methods inside

Reply:
should rename things into:
- `compute_optimal_batch_size`
- `_compute_batch_size_predictive`, maybe with no underscore
- `_compute_batch_size_bruteforce`, maybe with no underscore

Also, as there are now more methods, documentation should be moved from the Mixin to the specific methods